subtle_training_bugs.py

```
@@ -0,0 +1,787 @@
+
+    optimizer.step()
+
+
+def warm_restart_training():
+    model = nn.Linear(10, 2)
+    optimizer = optim.SGD(model.parameters(), lr=0.01)
+
+    data = torch.randn(32, 10)
+    target = torch.randint(0, 2, (32,))
+
+    checkpoint_path = "/tmp/checkpoint.pt"
```

Ruff (0.13.1)

Probable insecure usage of temporary file or directory: "/tmp/checkpoint.pt"

(S108)